File: ios/CareDoc/Sources/CareDoc/DesignSystem/Tokens.swift
import SwiftUI

// MARK: - Phantom design tokens（取自 phantom.com production CSS）

extension Color {
    init(hex: UInt32) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255
        )
    }

    /// 深淺色動態色
    init(light: UInt32, dark: UInt32) {
        #if os(iOS)
        self.init(uiColor: UIColor { trait in
            trait.userInterfaceStyle == .dark
                ? UIColor(Color(hex: dark))
                : UIColor(Color(hex: light))
        })
        #else
        self.init(nsColor: NSColor(name: nil) { appearance in
            appearance.bestMatch(from: [.darkAqua, .aqua]) == .darkAqua
                ? NSColor(Color(hex: dark))
                : NSColor(Color(hex: light))
        })
        #endif
    }
}

enum CD {
    // 背景與表面
    static let bg = Color(light: 0xFFFDF8, dark: 0x1C1C1C)
    static let surface = Color(light: 0xFDFCFE, dark: 0x28282C)
    static let surface2 = Color(light: 0xF4F2F4, dark: 0x34343A)
    static let surface3 = Color(light: 0xEDEDEF, dark: 0x2E2E32)

    // 文字
    static let text = Color(light: 0x3C315B, dark: 0xFFFDF8)
    static let text2 = Color(light: 0x86848D, dark: 0xA09FA6)
    static let text3 = Color(light: 0xA09FA6, dark: 0x86848D)

    // 品牌
    static let accent = Color(hex: 0xAB9FF2)       // 薰衣草紫
    static let accentSoft = Color(hex: 0xE2DFFE)
    static let plum = Color(hex: 0x3C315B)
    static let plumDeep = Color(hex: 0x2C2250)
    static let lemon = Color(hex: 0xF1FF52)
    static let cream = Color(hex: 0xFFFDF8)

    // 機能色
    static let success = Color(hex: 0x45E863)
    static let danger = Color(hex: 0xFF0037)
    static let caution = Color(hex: 0xFFD600)
    static let info = Color(hex: 0x6CA0FB)

    // 插畫粉色系（櫻花）
    static let pink = Color(hex: 0xFFB7D5)
    static let pinkLight = Color(hex: 0xFFD9E8)
    static let pinkHot = Color(hex: 0xFF7EB0)

    // 卡片邊框
    static let cardBorder = Color(light: 0x3C315B, dark: 0xFFFDF8).opacity(0.08)

    // 圓角
    static let rCard: CGFloat = 18
    static let rCardLarge: CGFloat = 24
    static let rRow: CGFloat = 14
    static let rIcon: CGFloat = 11

    // 動效（Phantom cubic-bezier(.22,1,.36,1) 0.4s）
    static let ease = Animation.timingCurve(0.22, 1, 0.36, 1, duration: 0.4)
    static let easeSlow = Animation.timingCurve(0.22, 1, 0.36, 1, duration: 0.7)
}

// MARK: - 字體（SF Pro Rounded 標題 + 系統內文）

extension Font {
    static func cdDisplay(_ size: CGFloat) -> Font {
        .system(size: size, weight: .heavy, design: .rounded)
    }
    static func cdTitle(_ size: CGFloat) -> Font {
        .system(size: size, weight: .bold, design: .rounded)
    }
    static func cdBody(_ size: CGFloat, weight: Font.Weight = .regular) -> Font {
        .system(size: size, weight: weight)
    }
}

File: flutter/carrius/lib/design/tokens.dart
import 'package:flutter/material.dart';

/// Phantom 設計 tokens（取自 phantom.com production CSS）。淺色為先。
/// 對應 SwiftUI 版 Tokens.swift，雙平台共用同一組數值。
class CD {
  // ---- 品牌（深淺共用） ----
  static const accent = Color(0xFFAB9FF2); // 薰衣草紫
  static const accentSoft = Color(0xFFE2DFFE);
  static const plum = Color(0xFF3C315B);
  static const plumDeep = Color(0xFF2C2250);
  static const lemon = Color(0xFFF1FF52);
  static const cream = Color(0xFFFFFDF8);

  // ---- 機能色 ----
  static const success = Color(0xFF45E863);
  static const danger = Color(0xFFFF0037);
  static const caution = Color(0xFFFFD600);
  static const info = Color(0xFF6CA0FB);

  // ---- 插畫粉色系 ----
  static const pink = Color(0xFFFFB7D5);
  static const pinkLight = Color(0xFFFFD9E8);
  static const pinkHot = Color(0xFFFF7EB0);

  // ---- 圓角 ----
  static const double rCard = 18;
  static const double rCardLarge = 24;
  static const double rRow = 14;
  static const double rIcon = 11;

  // ---- 動效（Phantom cubic-bezier(.22,1,.36,1)） ----
  static const ease = Cubic(0.22, 1, 0.36, 1);
  static const easeDur = Duration(milliseconds: 400);
  static const easeSlowDur = Duration(milliseconds: 700);

  // ---- 字體 ----
  static const display = 'Nunito';   // 圓潤標題（取代 SF Pro Rounded）
  static const body = 'NotoSansTC';  // 中文內文
}

/// 深淺色語意色票。預設淺色。
class Palette {
  final Color bg, surface, surface2, surface3;
  final Color text, text2, text3;
  final Color cardBorder;
  final Color tabbar;
  final Brightness brightness;

  const Palette({
    required this.bg,
    required this.surface,
    required this.surface2,
    required this.surface3,
    required this.text,
    required this.text2,
    required this.text3,
    required this.cardBorder,
    required this.tabbar,
    required this.brightness,
  });

  static const light = Palette(
    bg: Color(0xFFFFFDF8),
    surface: Color(0xFFFDFCFE),
    surface2: Color(0xFFF4F2F4),
    surface3: Color(0xFFEDEDEF),
    text: Color(0xFF3C315B),
    text2: Color(0xFF86848D),
    text3: Color(0xFFA09FA6),
    cardBorder: Color(0x143C315B), // plum 8%
    tabbar: Color(0xE0FDFCFE),
    brightness: Brightness.light,
  );

  static const dark = Palette(
    bg: Color(0xFF1C1C1C),
    surface: Color(0xFF28282C),
    surface2: Color(0xFF34343A),
    surface3: Color(0xFF2E2E32),
    text: Color(0xFFFFFDF8),
    text2: Color(0xFFA09FA6),
    text3: Color(0xFF86848D),
    cardBorder: Color(0x12FFFDF8), // cream 7%
    tabbar: Color(0xDB28282C),
    brightness: Brightness.dark,
  );
}

/// 讓 widget 取用目前色票。
class PaletteScope extends InheritedWidget {
  final Palette palette;
  const PaletteScope({super.key, required this.palette, required super.child});

  static Palette of(BuildContext context) {
    final scope = context.dependOnInheritedWidgetOfExactType<PaletteScope>();
    return scope?.palette ?? Palette.light;
  }

  @override
  bool updateShouldNotify(PaletteScope oldWidget) => oldWidget.palette != palette;
}

/// 字型工具（對應 SwiftUI .cdDisplay/.cdTitle/.cdBody）
class CDText {
  static TextStyle display(double size, {Color? color}) => TextStyle(
        fontFamily: CD.display,
        fontWeight: FontWeight.w900,
        fontSize: size,
        height: 1.15,
        letterSpacing: -0.5,
        color: color,
      );

  static TextStyle title(double size, {Color? color}) => TextStyle(
        fontFamily: CD.display,
        fontWeight: FontWeight.w800,
        fontSize: size,
        letterSpacing: -0.2,
        color: color,
      );

  static TextStyle body(double size, {FontWeight weight = FontWeight.w400, Color? color, double? height}) =>
      TextStyle(
        fontFamily: CD.body,
        fontWeight: weight,
        fontSize: size,
        height: height,
        color: color,
      );
}
